operator/internal/controller/vectorized/cluster_controller.go

```diff
@@ -80,6 +81,7 @@ type ClusterReconciler struct {
 	GhostDecommissioning      bool
 	AutoDeletePVCs            bool
 	Dialer                    redpanda.DialContextFunc
+	CloudExpander             *pkgsecrets.CloudExpander
```

Comment:
nit: in this context I think `SecretExpander` or `CloudSecretExpander` is more clear. Either way, it would hurt to add a comment noting what this is used for.